src/main/java/javaslang/collection/Iterator.java

```diff
@@ -1679,6 +1667,22 @@ public T next() {
 
     abstract class AbstractIterator<T> implements Iterator<T> {
 
+        public boolean equals(Object o) {
+            if(o instanceof java.lang.Iterable) {
```

Comment:
Throughout Javaslang equals is implemented as follows:

1. o instance of this.getClass(), i.e. o != null && this.getClass() is assignable from o.getClass()
2. fields are equal

To compare values only, one could do s.th. similar to Stream.ofAll(iter1).equals(Stream.ofAll(iter2)).